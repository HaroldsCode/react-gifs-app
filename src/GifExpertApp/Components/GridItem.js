import PropTypes from "prop-types";
export const GridItem = ({title, url}) => {
    return (
        <li className="card">
            <img className="card__img" src={url} alt={title} title={title}/>
        </li>
    );
}

GridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}