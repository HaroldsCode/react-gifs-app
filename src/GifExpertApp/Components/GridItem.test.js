import { shallow } from "enzyme"
import { GridItem } from "./GridItem";

describe('GridItem tests', () => {
    const title = 'titulito';
    const url = 'https://localhost/pp.jpg';
    const wrapper = shallow(<GridItem title={title} url={url}/>);

    test('should display the component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should have src and alt same to the props', () => {
        const image = wrapper.find('.card__img');
        expect(url).toBe(image.prop("src"));
        expect(title).toBe(image.prop("alt"));
    })
    
    test('should have card class', () => {
        const li = wrapper.find('li');
        expect(true).toBe(li.prop("className").includes('card'));
    })
})
