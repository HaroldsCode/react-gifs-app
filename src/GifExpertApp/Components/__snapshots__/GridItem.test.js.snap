// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem tests should display the component correctly 1`] = `
<li
  className="card"
>
  <img
    alt="titulito"
    className="card__img"
    src="https://localhost/pp.jpg"
    title="titulito"
  />
</li>
`;
